kani-compiler/src/codegen_cprover_gotoc/codegen/operand.rs

```diff
@@ -47,28 +47,37 @@ impl<'tcx> GotocCtx<'tcx> {
 
     fn codegen_constant(&mut self, c: &Constant<'tcx>) -> Expr {
         trace!(constant=?c, "codegen_constant");
-        let const_ = match self.monomorphize(c.literal) {
-            ConstantKind::Ty(ct) => ct,
-            ConstantKind::Val(val, ty) => return self.codegen_const_value(val, ty, Some(&c.span)),
-        };
+        let span = Some(&c.span);
+        match self.monomorphize(c.literal) {
+            ConstantKind::Ty(ct) => self.codegen_const(ct, span),
+            ConstantKind::Val(val, ty) => self.codegen_const_value(val, ty, span),
+            ConstantKind::Unevaluated(unevaluated, ty) => {
+                self.codegen_const_unevaluated(unevaluated, ty, span)
+            }
+        }
+    }
 
-        self.codegen_const(const_, Some(&c.span))
+    fn codegen_const_unevaluated(
+        &mut self,
+        unevaluated: UnevaluatedConst<'tcx>,
+        ty: Ty<'tcx>,
+        span: Option<&Span>,
+    ) -> Expr {
+        debug!("The literal was a Unevaluated");
```

Comment:
Can you please add the constant?

```rust
debug!(?unevaluated, "codegen_const_unevaluated");
```

Nit: I usually add the name of the function to the log so it's easy to find it in the code. But I don't think we have any guidelines on this. So just a suggestion.